src/commands/force/source/retrieve.ts

```diff
@@ -220,4 +229,56 @@ export class Retrieve extends SourceCommand {
     });
     return hasCustomField && !hasCustomObject;
   }
+
+  private async moveResultsForRetrieveTargetDir(): Promise<void> {
+    async function mv(src: string): Promise<string[]> {
+      let directories: string[] = [];
+      let files: string[] = [];
+      const srcStat = await fs.promises.stat(src);
+      if (srcStat.isDirectory()) {
+        const contents = await fs.promises.readdir(src);
+        directories = contents.filter((c) => fs.statSync(join(src, c)).isDirectory()).map((c) => join(src, c));
+        files = contents.filter((c) => !fs.statSync(join(src, c)).isDirectory());
+      } else {
+        files.push(src);
+      }
+      await promisesQueue(
+        files,
+        async (file: string): Promise<string> => {
+          const dest = join(src.replace(join('main', 'default'), ''), file);
+          const destDir = dirname(dest);
+          await fs.promises.mkdir(destDir, { recursive: true });
+          await fs.promises.rename(join(src, file), dest);
+          return dest;
+        },
+        50
+      );
+      return directories;
+    }
+
+    if (!this.flags.retrievetargetdir) {
+      return;
+    }
+    let overlapsProject = true;
+    const resolvedTargetDir = resolve(this.flags.retrievetargetdir as string);
+    overlapsProject = !!this.project.getPackageDirectories().find((pkgDir) => {
+      if (pkgDir.fullPath) {
+        return pkgDir.fullPath.includes(resolvedTargetDir);
+      }
+      return false;
+    });
+    if (overlapsProject) {
+      return;
```

Comment:
this feels like a good place for a warning.  Another option would be to make this "overlaps" validation a standalone function that could be used for your new flag's `parse` so that it doesn't go through all that retrieve work to fail this late in the game.

As is, it looks like the user would specify a targetDir and then if it overlapped, the retrieve would go to their default dir anyway without a warning?

Reply:
@mshanemc I think I am going to change direction on overlapping targets, fail the command if the target overlaps a package dir. There are other commands and variants of source:retrieve to get source into a package dir.